src/components/widget/BarcodeScanner/BarcodeScannerWidget.js

```diff
@@ -0,0 +1,77 @@
+import React, { Component } from 'react';
+import BarcodeScanner, { BarcodeScannerResult } from './BarcodeScanner';
+
+function addBarcodeScanner(WrappedComponent) {
+  return class BarcodeScannerWidget extends Component {
+    constructor(props) {
+      super(props);
+
+      this.state = {
+        scanning: false,
+        result: null,
+        barcodeSelected: null,
+      };
+    }
+
+    _scanBarcode = val => {
```

Comment:
Does the leading underscore have any implication here or could it be removed?

Reply:
It's a leftover from when it was not a standalone HOC.